src/h3lib/lib/h3UniEdge.c

```diff
@@ -179,7 +179,7 @@ int H3_EXPORT(h3UnidirectionalEdgeIsValid)(H3Index edge) {
         return 0;
     }
 
-    return H3_EXPORT(h3IsValid)(origin);
+    return H3_EXPORT(h3IsValid)(origin & H3_RESERVED_MASK_NEGATIVE);
```

Comment:
Why do we need this change? `getOriginH3IndexFromUnidirectionalEdge` should already return a valid `origin` hex with the reserved bits properly set to zero.

Reply:
I think @isaacbrodsky meant to run this on the original `edge` index passed in, instead of on the generated `origin`.

Reply:
Yes, this is actually a no-op. Removed the mask, and rely on a test to ensure the highest bit is passed through and checked in `h3IsValid`.